_posts/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins.md

```diff
@@ -0,0 +1,336 @@
+      fullWidth={false}
+      onSelectedDataSource={(ds) => setSelectedDataSources(ds)}
+      disabled={false}
+    />
+  );
+```
+Then add an `onClickHandler` to retrieve indexes based on the selected data source:
+```
+ # public/components/app.tsx   
+   const [indices, setIndices] = useState<string[]>([]);
+   const onClickHandler = async () => {
+     if (selectedDataSources.length === 0) {
+       const resp = await http.get(`/api/blog_example_plugin/example`, { query: {} } );
+       setIndices(resp);
+    } else {
+      const dataSourceID =selectedDataSources[0].id;
+      // Use the core http service to make a response to the server API.
+      const resp = await http.get(`/api/blog_example_plugin/example`, {
+        query: {
+          dataSourceId: dataSourceID,
+        }
+      });
+      setIndices(resp);
+    }
+  };
+```
+Now, combine the preceding function and react component:
+```
+retrun (
+    ...
+   <EuiHorizontalRule />
+   {DataSourceSelectorComponent}
+   <EuiHorizontalRule />
+     <EuiTextColor>
+       Selected: {selectedDataSources.map((ds) => ds.label).join(', ')}
+   </EuiTextColor>
+   <EuiHorizontalRule />
+   <EuiButton type="primary" size="s" onClick={onClickHandler}>
+     <FormattedMessage id="mayTest.buttonText" defaultMessage="Get Indices" />
+   </EuiButton>
+   <p>
+     <FormattedMessage
+       id="blogTest.indicesText"
+       defaultMessage="Get indices: {data}"
+       values={{ data: indices ? indices : 'Unknown' }}
+     />
+   </p>
+  ...
+)
+```
+
+Congratulations! Now you can run your plugins and use the pickers to select the desired data source. 
+To get indexes from a remote cluster, call `datasource1`, as shown in the following image:
+<img src="/assets/media/blog-images/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins/blog_example_remote_cluster.png">
+
+
+Refer to the [blog_example_plugin](https://github.com/zhyuanqi/OpenSearch-Dashboards/tree/blog_example/examples/blog_example_plugin) repository for additional details about the example plugin code. 
+
+## Developing, testing, and debugging
+
+Now that you've learned how to integrate multiple data sources with OpenSearch Dashboards plugins, let's explore some advanced concepts and techniques.
+
+### Example plugins for multiple data sources
+
+To provide a more intuitive and direct experience, we've included an example plugin for developers to get started quickly with the components. You can enable the example plugins by running OpenSearch Dashboards with the following CLI command:
+```
+yarn start --run-examples
+```
+
+With OpenSearch Dashboards running, navigate to the page and select the left menu. Under OpenSearch Dashboards, open **Developer Examples**, then select **Multiple Data Sources Integration**, as shown in the following image. On the right side, you'll find a list of pickers, each with a brief introduction and properties listed below. Feel free to experiment with the example plugin.
```

Comment:
Second sentence: Either "Under **OpenSearch Dashboards**, or the sentence should just start with "Open **Developer Examples**..."

Reply:
change to `Open Developer examples, ...`